apps/angular/src/app/features/anime/anime-table/anime-table.component.ts

```diff
@@ -0,0 +1,148 @@
+
+  @ViewChild(MatPaginator) private readonly paginator: MatPaginator;
+
+  @ViewChild(MatSort) private readonly sort: MatSort;
+
+  /** Current anime list. */
+  public readonly animeData$: Observable<Pagination<Anime>>;
+
+  /** Current page. */
+  public readonly currentPage$ = new BehaviorSubject<number>(0);
+
+  /** Sorting. */
+  public readonly sorting$ = new BehaviorSubject<Sort>({
+    active: '',
+    direction: '',
+  });
+
+  /** Maximum anime in page. */
+  public maximumAnimeOnPage = 10;
+
+  /** Filtering field form controller. */
+  public filterFormControl = new FormControl<AnimeType[]>([], {
+    nonNullable: true,
+  });
+
+  /** Searching input form controller. */
+  public searchFormControl = new FormControl('', {
+    nonNullable: true,
+  });
+
+  public constructor(
+    animeService: AnimeService,
+    private readonly searchParamsService: SearchParamsService,
+  ) {
+    const {
+      maximumItemsOnPage,
+      pageNumber,
+      sorting,
+      searchingTitlePart,
+      types,
+    } = this.searchParamsService.getAnimeListSearchParams();
+
+    this.maximumAnimeOnPage = maximumItemsOnPage;
+    this.currentPage$.next(pageNumber);
+    this.sorting$.next(sorting);
+    this.searchFormControl.setValue(searchingTitlePart);
+    this.filterFormControl.setValue(types);
+
+    const filterChanges$ = this.filterFormControl.valueChanges.pipe(
+      startWith(this.filterFormControl.value),
+    );
+    const searchChanges$ = this.searchFormControl.valueChanges.pipe(
+      startWith(this.searchFormControl.value),
+    );
+    const sortingChanges$ = this.sorting$;
+
+    const resetPaginationChanges$ = combineLatest([
+      filterChanges$,
+      searchChanges$,
+      sortingChanges$,
+    ]).pipe(
+      tap(() => {
+        this.currentPage$.next(0);
+      }),
+    );
+
+    this.animeData$ = resetPaginationChanges$.pipe(
+      combineLatestWith(this.currentPage$),
+      debounceTime(300),
+      switchMap(([[selectedFilter, selectedSearch, selectedSorting], selectedPage]) => {
+        const params = this.searchParamsService.changeSearchParams(new AnimeListSearchParams({
+          maximumItemsOnPage: this.maximumAnimeOnPage,
+          pageNumber: selectedPage,
+          sorting: selectedSorting,
+          types: selectedFilter,
+          searchingTitlePart: String(selectedSearch),
+        }));
+        return animeService.getAnimeList(params);
+      }),
+    );
+  }
+
+  /**
+   * Initialize paginator.
+   * @inheritdoc
+   */
+  public ngAfterViewInit(): void {
+    this.dataSource.paginator = this.paginator;
+    this.dataSource.sort = this.sort;
+  }
+
+  /**
+   * Changes sorting target.
+   * @param sorting Sort event.
+   */
+  public changeSortingTarget(sorting: Sort): void {
+    this.sorting$.next(sorting);
+  }
+
+  /**
+   * Changes current page.
+   * @param event Page event.
+   */
+  public changePage(event: PageEvent): void {
```

Comment:
We use `on` prefix for event handlers on Angular projects
```suggestion
  public onPageChange(event: PageEvent): void {
```